File: concurrency/concurrent.go
package main

import (
	"fmt"
	"sync"
	"time"
)

func printNumbers(start, end int, wg *sync.WaitGroup) {
	defer wg.Done() // Mark this goroutine as done when function returns

	for i := start; i <= end; i++ {
		fmt.Printf("Number: %d\n", i)
		time.Sleep(100 * time.Millisecond) // Small delay to simulate work
	}
}

func main() {
	var wg sync.WaitGroup
	wg.Add(2) // We will launch 2 goroutines

	// Launch goroutines
	go printNumbers(1, 5, &wg)
	go printNumbers(6, 10, &wg)

	// Wait for goroutines to complete
	wg.Wait()

	fmt.Println("Go goroutines completed")
}

File: concurrency/concurrent.py
import threading
import time


def print_numbers(start, end):
    for i in range(start, end + 1):
        print(f"Number: {i}")
        time.sleep(0.1)  # Small delay to simulate work


# Create threads
t1 = threading.Thread(target=print_numbers, args=(1, 5))
t2 = threading.Thread(target=print_numbers, args=(6, 10))

# Start threads
t1.start()
t2.start()

# Wait for threads to complete
t1.join()
t2.join()

print("Python threading completed")
